r/configure

```diff
@@ -293,15 +293,15 @@ set_pkg_vars () {
 
 # If we have pkg-config, it will tell us what libarrow needs
 set_lib_dir_with_pc () {
-  LIB_DIR="`${PKG_CONFIG} --variable=libdir --silence-errors ${PKG_CONFIG_NAME}`"
+  LIB_DIR="`${PKG_CONFIG} --variable=libdir  ${PKG_CONFIG_NAME}`"
```

Comment:
If there are no errors there is no ouput with erros it looks like this:
```
dewey@tidytuesday ~ % R CMD INSTALL ~/Desktop/arrow_14.0.0.tar.gz 
* installing to library ‘/Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library’
* installing *source* package ‘arrow’ ...
** using staged installation
*** Found libcurl and OpenSSL >= 1.1
*** Successfully retrieved C++ binaries (darwin-x86_64-openssl-1.1)
*** Checksum validated successfully for libarrow binary: darwin-x86_64-openssl-1.1/arrow-14.0.0.zip
Package libcurl was not found in the pkg-config search path.
Perhaps you should add the directory containing `libcurl.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libcurl', required by 'arrow', not found
Package libcurl was not found in the pkg-config search path.
Perhaps you should add the directory containing `libcurl.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libcurl', required by 'arrow', not found
Package libcurl was not found in the pkg-config search path.
Perhaps you should add the directory containing `libcurl.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libcurl', required by 'arrow', not found
```

Reply:
(this does not kill the script)

Reply:
Ok, thanks. IMO that's pretty noisy (and it all sounds very bad ™️ to someone who isn't used to this, which will be confusing for them since it almost certainly isn't the error if they run into it) so I would advocate for adding the silencing back in as soon as we can.

If you think it's better to leave it for just this one release that's ok, but if I'm reading this correctly we should get a good sign that this has happened from line https://github.com/apache/arrow/pull/38970/files#diff-089697faebdb7820ca629a2bb316b878cc0ba18a5bfb0b60996f8dbcd1fa11e7R407 right after with a much nicer message, so maybe we don't really need it in this release? Regardless, we should plan to add it back in next release at the latest (and if we're going to do that, we should make an issue / add a comment, yeah?).

Reply:
> and there's no indication that this type of ouput would occur and lead to success anywhere except a CRAN MacOS build

I'm honestly more concerned with the case where this output happens and the build fails elsewhere and that confuses people as to what the real issue is than this popping up + it succeeding.

I hear the concern about development time, and for that IMO we should strive to replicate CRAN environments when we need to (and like we've done) + put this behind debug flags so we can see it when we need it (or even as default in our CI) but not expose this to users. These messages will only confuse people who see them outside of us.

Reply:
> I'm honestly more concerned with the case where this output happens and the build fails elsewhere and that confuses people as to what the real issue is than this popping up + it succeeding.

From my experience people will not even see that, they just see the cmake error (which is likely the last output if the build proceeds even for some reason e.g. #38769) and open an issue. In addition I don't really want to have to do a 'turn on debug options on cran so we can pinpoint the issue'-submissions anymore out of respect for the cran resources as well as my time waiting for the check to be finally run (which took several days this time around) and of course the time of nic/the maintainer to do the submission(especially now that the 'need' for secrecy is gone).

And while theoretically reproducing the CRAN environment would work, there is just not enough info to do that as we have seen with this issue. (I would of course love to be able to do that but even having an actual 10.13 runner didn't protect us...)

Reply:
But we didn't use these messages on CRAN to diagnose anything (and frankly I don't think had we generally seen them we would have known what was going on to fix if they had been there in the first place), so I'm skeptical about the amount they will help versus be noise that is confusing.

If we're going to have them be there, at the very least could we collect them up and then dump them only if there's a failure in this process (+describe that they are useful only to maintainers)?